src/lexers/jsx-lexer.js

```diff
@@ -34,18 +34,23 @@ export default class JsxLexer extends HTMLLexer {
     return this.keys
   }
 
+  /**
+  * Extract tags and content from the Trans component.
+  * @param {string} string
+  * @returns {array} Array of key options
+  */
   extractTrans(content) {
     let matches
-    const closingTagPattern = '(?:<Trans([^>]*\\s' + this.attr + '[^>]*?)\\/>)'
-    const selfClosingTagPattern = '(?:<Trans([^>]*\\s' + this.attr + '[^>]*?)>((?:\\s|.)*?)<\\/Trans>)'
+    const selfClosingTagPattern = '(?:<\\s*Trans([^>]*)?/>)'
+    const closingTagPattern = '(?:<\\s*Trans([^>]*)?>((?:(?!</\\s*Trans\\s*>)[^])*)</\\s*Trans\\s*>)'
     const regex = new RegExp(
-      [closingTagPattern, selfClosingTagPattern].join('|'),
+      [selfClosingTagPattern, closingTagPattern].join('|'),
       'gi'
     )
 
     while (matches = regex.exec(content)) {
       const attrs = this.parseAttributes(matches[1] || matches[2])
-      const key = attrs.keys
+      const key = attrs.keys || matches[3]
```

Comment:
If we didn't get a key back from `parseAttributes`, use the content we matched (which is `undefined` if we didn't match it or `''` if we matched nothing -- both of which are falsey.)